bundles/org.openhab.binding.bluetooth.bluez/src/main/java/org/openhab/binding/bluetooth/bluez/internal/BlueZBridgeHandler.java

```diff
@@ -0,0 +1,241 @@
+     */
+    public BlueZBridgeHandler(Bridge bridge, DeviceManagerFactory deviceManagerFactory) {
+        super(bridge);
+        this.deviceManagerFactory = deviceManagerFactory;
+    }
+
+    @Override
+    public void initialize() {
+        super.initialize();
+
+        // Load configuration
+        final BlueZAdapterConfiguration configuration = getConfigAs(BlueZAdapterConfiguration.class);
+        String addr = configuration.address;
+        if (addr != null) {
+            this.adapterAddress = new BluetoothAddress(addr.toUpperCase());
+        } else {
+            // If configuration does not contain adapter address to use, exit with error.
+            logger.info("Adapter MAC address not provided");
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "address not set");
+            return;
+        }
+
+        logger.debug("Creating BlueZ adapter with address '{}'", adapterAddress);
+
+        updateStatus(ThingStatus.UNKNOWN, ThingStatusDetail.CONFIGURATION_PENDING, "Initializing");
+
+        deviceManagerFactory.getPropertiesChangedHandler().addListener(this);
+
+        discoveryJob = scheduler.scheduleWithFixedDelay(this::initializeAndRefreshDevices, 5, 10, TimeUnit.SECONDS);
+    }
+
+    @Override
+    public void dispose() {
+        deviceManagerFactory.getPropertiesChangedHandler().removeListener(this);
+        logger.debug("Termination of DBus BlueZ handler");
+
+        Future<?> job = discoveryJob;
+        if (job != null) {
+            job.cancel(false);
+            discoveryJob = null;
+        }
+
+        BluetoothAdapter localAdatper = this.adapter;
+        if (localAdatper != null) {
+            localAdatper.stopDiscovery();
+            this.adapter = null;
+        }
+
+        super.dispose();
+    }
+
+    private @Nullable BluetoothAdapter prepareAdapter(DeviceManagerWrapper deviceManager) {
+        // next lets check if we can find our adapter in the manager.
+        BluetoothAdapter localAdapter = adapter;
+        if (localAdapter == null) {
+            BluetoothAddress localAddress = adapterAddress;
+            if (localAddress != null) {
+                localAdapter = adapter = deviceManager.getAdapter(localAddress);
+            } else {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "No adapter address provided");
+                return null;
+            }
+        }
+        if (localAdapter == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    "Native adapter could not be found for address '" + adapterAddress + "'");
+            return null;
+        }
+        // now lets confirm that the adapter is powered
+        if (!localAdapter.isPowered()) {
+            localAdapter.setPowered(true);
+            // give the device some time to power on
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.DUTY_CYCLE,
```

Comment:
As far as I know, there isn't such a thing as duty cycle for BT - that's only a regulation on 868/900MHz. Better put detail to NONE.

Reply:
changed to NONE

Reply:
@kaikreuzer it is not because of regulation, but with testing, what I've realized is that when you power on the device, if you immediately start the discovery, it might fail, so you need to give the adapter a little time to start up. But yes the "DUTY_CYCLE" is not the right reason.